blog/the-big-secret-of-react.md

```diff
@@ -0,0 +1,48 @@
+# The big secret of React
+
+Did you wonder how React compares the `deps` array used by the hooks in every render? Let's check the magic that decides when run the `useEffect` hook.
+
+## Introduction
+
+As a curious developer, I was wondering how React checks the array of dependencies, called `deps`, that I pass to the `useEffect`, `useCallback` and `useMemo` hooks. I didn't find the answer anywhere.
+
+```javascript
+useEffect(() => {
+  console.log('Hook was executed');
+}, [someValue]);
+```
+
+So, my unique option was to check the React's code itself to find the magic, and I found more than I expected.
+
+## The comparison
+
+React don't use `==` or `===` operators, it uses the [`Object.is()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) method. This method provides the [`SameValue` algorithm](https://262.ecma-international.org/5.1/#sec-9.12) since ES2015.
```

Comment:
```suggestion
React doesn't use `==` or `===` operators, it uses the [`Object.is()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) method. This method provides the [`SameValue` algorithm](https://262.ecma-international.org/5.1/#sec-9.12) since ES2015.
```